Add PHPUnit tests for links_add_base_url function

pbearne opened the pull request:
A new PHPUnit test class, Tests_formatting_linksAddBaseUrl, has been introduced to increase the test coverage for the links_add_base_url function. The class includes multiple test scenarios, examining different combinations of attributes to verify the correct functioning of the function.


Trac ticket: https://core.trac.wordpress.org/ticket/60389

pbearne committed:
Add PHPUnit tests for links_add_base_url function

A new PHPUnit test class, Tests_formatting_linksAddBaseUrl, has been introduced to increase the test coverage for the links_add_base_url function. The class includes multiple test scenarios, examining different combinations of attributes to verify the correct functioning of the function.

pbearne committed:
Update linksAddBaseUrl.php

pbearne committed:
Update linksAddBaseUrl.php

aaronjorbin commented on tests/phpunit/tests/formatting/linksAddBaseUrl.php:
```diff
@@ -0,0 +1,67 @@
+	}
+
+	public function data_links_add_base_url() {
+		return array(
```
Some additional cases that I think need to be covered:

- `http` rather than `https`
- making sure that unless `data-url` is set in the `attrs`, it isn't adjusted
- the same as what is the current `data_url` but with one of them setup properly
- no protocol URLs present in content (`//localhost/url` )
- URLs that use other protocols like FTP and mailto
- A data URL in one of the attributes

I have no idea what the results for some of these are going to be.

pbearne replied:
added some more test

pbearne committed:
Add more test scenarios in linksAddBaseUrl.php

Expanded test cases in 'linksAddBaseUrl.php' to include cases for 'http', 'ftp' and 'ftps' bases in addition to the existing cases. Breadth of testing is enhanced with this consideration of more URL schemas.

peterwilsoncc committed:
Run phpcbf.

peterwilsoncc committed:
Merge branch 'trunk' into 60389-links_add_base_url

peterwilsoncc commented on tests/phpunit/tests/formatting/linksAddBaseUrl.php:
```diff
@@ -0,0 +1,83 @@
+
+	public function data_links_add_base_url() {
+		return array(
+			'default'      => array(
```
```suggestion
			'https'      => array(
```

peterwilsoncc commented on tests/phpunit/tests/formatting/linksAddBaseUrl.php:
```diff
@@ -0,0 +1,83 @@
+				'base'     => 'http://localhost',
+				'attrs'    => null,
+				'expected' => '<a href="http://localhost/url" />',
+			),
```
```suggestion
			),
			'relative_scheme'         => array(
				'content'  => '<a href="//localhost/url" />',
				'base'     => 'http://localhost',
				'attrs'    => null,
				'expected' => '<a href="http://localhost/url" />',
			),
```

peterwilsoncc reviewed (commented):
Naming things and a realtive scheme test.

Please confirm any coding stanadards and results.

pbearne committed:
Update tests/phpunit/tests/formatting/linksAddBaseUrl.php

Co-authored-by: Peter Wilson <519727+peterwilsoncc@users.noreply.github.com>

pbearne committed:
Update tests/phpunit/tests/formatting/linksAddBaseUrl.php

Co-authored-by: Peter Wilson <519727+peterwilsoncc@users.noreply.github.com>

mukeshpanchal27 reviewed (changes requested):
@pbearne Needs to fix the PHPCS fix.

pbearne committed:
Update linksAddBaseUrl.php

whitespace fixes

pbearne committed:
Update linksAddBaseUrl.php

pbearne committed:
Update linksAddBaseUrl.php

pbearne commented:
> @pbearne Needs to fix the PHPCS fix.

fixed

aristath reviewed (approved):
LGTM 👍

peterwilsoncc committed:
Merge branch 'trunk' into 60389-links_add_base_url

peterwilsoncc committed:
Merge branch 'trunk' into 60389-links_add_base_url

peterwilsoncc committed:
Dataprovder docblock

peterwilsoncc commented:
Merged [r59162](https://core.trac.wordpress.org/changeset/59162) / https://github.com/WordPress/wordpress-develop/commit/636d748a4936216a1567d0898d8fa15a08f415b0
